doc/ug/bazel_notes.md

```diff
@@ -334,6 +342,19 @@ For example, to run all chip-level tests except the broken ones on FPGA:
 bazel test --test_tag_filters=cw310,-broken //sw/device/tests/...
 ```
 
+## Using Bazel with Git Worktrees
+
+Bazel was not optimized for the `git` worktree workflow, but using worktrees can help with branch management and provides the advantage of being able to run multiple Bazel jobs simultaneously.
```

Comment:
nit: Do we have a recommended `worktree` setup? Many contributors are not necessarily familiar with git.

Reply:
This is how I have things setup: https://morgan.cugerone.com/blog/how-to-use-git-worktree-and-in-a-clean-way/ which I have found convenient

Reply:
but short answer: we don't have a recommended way in any documentation I don't think

Reply:
We don't currently have a recommendation for whether or not to use worktrees. It's just that a subset of the people on the team seem to use it, so I thought it might be good to document some of the issues I ran into. The [disk cache](https://docs.opentitan.org/doc/ug/bazel_notes/#disk-cache) section of this page has a link to Git's documentation of the feature. This same advice actually also applies if anyone on the team is cloning the repository multiple times in different directories to checkout multiple branches.